# Linear Regression using sklearn
Find a Python library that computes the optimal coefficients in the linear regression
problem, and compare its solution with Theano's Linear Regression.

In [3]:
import numpy as np
from sklearn.linear_model import LinearRegression

In [4]:
# load csv into ndarray
data = np.genfromtxt('linear.csv', delimiter=',')

## 1. Process Data

In [5]:
# extract feature matrix and labels
features = data[:,1:]
labels = data[:,0]

## 2. Initialize sklearn's LinearRegression Class

In [12]:
# init the regressor
regressor = LinearRegression()

## 3. Compute Coefficients
Call `regressor.fit()` and print the `coef_` and `intercept_`.

In [17]:
regressor.fit(features, labels)
print(regressor.coef_)
print(regressor.intercept_)

[-0.57392068  1.35757059  0.01527565  0.        ]
-1.8828807643601513


## 4. Compare Results
From Theano, we got weights

```
[[-0.57392068]
 [ 1.35757059]
 [ 0.01527565]
 [-1.88288076]]
```

`sklearn` separates these into the learned coefficients, `coeff_`, $w_{3}, w_{2}, w_{1}$, `= [-0.57392068  1.35757059  0.01527565  0.        ]` and independent term `intercept_` $w_{0}$, `= -1.8828807643601513`.
